server/interview/views.py

```diff
@@ -17,71 +18,95 @@
 
 logger = logging.getLogger(__name__)
 
+def parse_date(x):
+    date = datetime.strptime(x, "%Y-%m-%dT%H:%M:%S.%fZ")
+
+    if timezone.is_aware(date):
+        return date
+    return timezone.make_aware(date)
+
 # custom permission to only allow participants of an interview to view it.
 class IsInterviewParticipant(permissions.BasePermission):
     def has_object_permission(self, request, view, obj):
         # Check if the user is either the interviewer or interviewee
         return obj.interviewer == request.user or obj.interviewee == request.user
 
+
 # Helper validation functions
 def is_valid_availability(availability):
-    return isinstance(availability, list) and len(availability) == 48 and all(isinstance(slot, bool) for slot in availability)
+    return (
+        isinstance(availability, list)
+        and len(availability) == 48
+        and all(isinstance(slot, bool) for slot in availability)
+    )
+
 
 # Create your views here.
 class AuthenticatedMemberSignupForInterview(APIView):
     permission_classes = [IsAuthenticated]
 
     def get(self, request):
-        logger.debug(f"GET request received for AuthenticatedMemberSignupForInterview by user {request.user.username}")
+        logger.debug(
```

Comment:
You shouldn't use f-strings with logger as it will always call `str()` on the f-string when passing over it even when we aren't in debug mode etc. Its better to use `logger.debug("GET request received for AuthenticatedMemberSignupForInterview by user %s", request.user.username)` as that won't be executed if we aren't in the mode.

Reply:
Oh yeah good call. We've definitely discussed this before too lol

Reply:
Thanks :)